src/analyzer/analyzermanager.cpp

```diff
@@ -0,0 +1,295 @@
+#include "analyzer/analyzermanager.h"
+
+#include <typeinfo>
+#include <QThread>
+
+#include <QtDebug>
+#include <QMutexLocker>
+#include <QListIterator>
+
+#include "library/trackcollection.h"
+#include "mixer/playerinfo.h"
+#include "track/track.h"
+#include "util/compatibility.h"
+#include "util/event.h"
+#include "util/timer.h"
+#include "util/trace.h"
+
+AnalyzerManager::AnalyzerManager(UserSettingsPointer pConfig) :
+    m_pConfig(pConfig),
+    m_nextWorkerId(0),
+    m_batchTrackQueue(),
+    m_prioTrackQueue(),
+    m_backgroundWorkers(),
+    m_foregroundWorkers(),
+    m_pausedWorkers() {
+
+    int maxThreads = m_pConfig->getValue<int>(ConfigKey("[Library]", "MaxAnalysisThreads"));
+    int ideal = QThread::idealThreadCount();
+    if (QThread::idealThreadCount() < 1) {
```

Comment:
If idealthreadcount is -1 (or generally, less than 1), then the program can do two things:
a)Accept what the user configured in the configuration
b)Use only one thread.

I do "b" if the configured value does not seem sane, and do a if it seems sane.

On the preferences dialog, there is a code similar to this, but with the difference that, if idealthreadcount is -1 ( < 1), the user can select between 1 and 8, and the default value is 1.